Fix transaction_tests.rs compilation errors: missing conflict_clause field

## Bug Description

The `transaction_tests.rs` file has compilation errors due to missing `conflict_clause` field in `InsertStmt` initializations. This appears to be a pre-existing issue that is blocking test compilation in multiple PRs.

## Error

```
error[E0063]: missing field `conflict_clause` in initializer of `InsertStmt`
   --> crates/vibesql-executor/src/tests/transaction_tests.rs:127:23
    |
127 |     let insert_stmt = InsertStmt {
    |                       ^^^^^^^^^^ missing `conflict_clause`
```

This error occurs 11 times across the transaction_tests.rs file.

## Files Affected

- `crates/vibesql-executor/src/tests/transaction_tests.rs` (11 occurrences)

## Impact

- **Severity**: Medium (blocks lib test compilation but not main binary)
- **Scope**: Test infrastructure only
- **Workaround**: Use `--bin` tests or compile without tests

## Required Fix

Add `conflict_clause: None` (or appropriate value) to all `InsertStmt` initializations in transaction_tests.rs:
- Lines 127, 159, 188, 202, 230, 244, 279, 314, 343, 371, 384

## Context

Discovered during review of PR #1322. The trigger implementation PR itself is clean, but cannot run lib tests due to this pre-existing issue.

--- a/crates/vibesql-executor/src/tests/transaction_tests.rs
+++ b/crates/vibesql-executor/src/tests/transaction_tests.rs
@@ -131,6 +131,7 @@ fn test_transaction_insert_commit() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     let rows = InsertExecutor::execute(&mut db, &insert_stmt).unwrap();
     assert_eq!(rows, 1);
@@ -163,6 +164,7 @@ fn test_transaction_insert_rollback() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     let rows = InsertExecutor::execute(&mut db, &insert_stmt).unwrap();
     assert_eq!(rows, 1);
@@ -192,6 +194,7 @@ fn test_transaction_multiple_operations_commit() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt1).unwrap();
 
@@ -206,6 +209,7 @@ fn test_transaction_multiple_operations_commit() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(2)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Bob".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt2).unwrap();
 
@@ -234,6 +238,7 @@ fn test_transaction_multiple_operations_rollback() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt1).unwrap();
 
@@ -248,6 +253,7 @@ fn test_transaction_multiple_operations_rollback() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(2)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Bob".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt2).unwrap();
 
@@ -283,6 +289,7 @@ fn test_transaction_isolation() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db1, &insert_stmt).unwrap();
 
@@ -318,6 +325,7 @@ fn test_transaction_nested_operations() {
                 vibesql_ast::Expression::Literal(SqlValue::Integer(i)),
                 vibesql_ast::Expression::Literal(SqlValue::Varchar(format!("User{}", i))),
             ]]),
+            conflict_clause: None,
         };
         InsertExecutor::execute(&mut db, &insert_stmt).unwrap();
     }
@@ -347,6 +355,7 @@ fn test_transaction_empty_rollback() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt).unwrap();
 
@@ -375,6 +384,7 @@ fn test_multiple_transactions() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(1)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Alice".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt1).unwrap();
     CommitExecutor::execute(&CommitStmt, &mut db).unwrap();
@@ -388,6 +398,7 @@ fn test_multiple_transactions() {
             vibesql_ast::Expression::Literal(SqlValue::Integer(2)),
             vibesql_ast::Expression::Literal(SqlValue::Varchar("Bob".to_string())),
         ]]),
+        conflict_clause: None,
     };
     InsertExecutor::execute(&mut db, &insert_stmt2).unwrap();
     RollbackExecutor::execute(&RollbackStmt, &mut db).unwrap();

